### PR TITLE
Update Release Schedule (1.11)

### PR DESCRIPTION
**Pull request by skonto**
# Changes

- Similar to https://github.com/knative/community/pull/1336
- Technically 1.9 is EOL tomorrow.

### CODE REVIEW

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Review by psschwei — COMMENTED** [review]
Should we drop v1.10 and v1.11 from the upcoming releases section?

File: mechanics/RELEASE-SCHEDULE.md [hunk 1]
```diff
@@ -22,8 +22,6 @@ With that said, it can be useful to have a list of when future releases will hap
 
 | Release | Date       | EOL        | Min K8s Version | Notes |
 | ------- | ---------- | ---------- | --------------- | ----- |
-| 1.10    | 2023-04-25 | 2023-10-31 | 1.24            |       |
-| 1.11    | 2023-07-25 | 2024-01-30 | 1.25            |       |
 | 1.12    | 2023-10-24 | 2024-04-30 | 1.26            |       |
 | 1.13    | 2024-01-23 | 2024-07-30 | -               |       |
```
Comment (skonto) [inline mechanics/RELEASE-SCHEDULE.md:26]: Do we know what min version we want to support for 1.13+?

Comment (dsimansk) [reply mechanics/RELEASE-SCHEDULE.md:26]: Likely `1.27` it seems to me.

**Pull request merged**